### ISSUE TITLE
Cinnamon 2.2 and GTK+ 3.14

### ISSUE THREAD

**Issue by dodev** [rbrito/pkg-mediterranean-gtk-themes#15]
Hi!

I'm a Mediterranean theme user for a while now and it is by far the cleanest, neatest and sexiest theme out there! Kudos!

But recently I upgraded to cinnamon from 2.0 to 2.2 and it turns out it breaks parts of the Mediterranean wiget and window themes. I checked my version of GTK+ and it's 3.14.1.

Here are some of the things that are broken:
![New window decoration](https://cloud.githubusercontent.com/assets/1612196/4703855/fdec6ff4-586d-11e4-88aa-dbba90fc1785.jpg)
![Button groups](https://cloud.githubusercontent.com/assets/1612196/4703857/fdf252c0-586d-11e4-9c98-72a802869a9f.jpg)
![No checkboxes](https://cloud.githubusercontent.com/assets/1612196/4703856/fdf0d076-586d-11e4-96e1-19c768f740e9.jpg)

Is there something I can do about this?

**Comment by rbrito** [comment]
Dear @dodev.

Can you please check if you still have issues with the newest changes that were contributed by @robertpainsi, among other people?

BTW, if you know some CSS, we desperately need some help to refactor and make these themes maintainable.

Once you confirm that this issue is gone, I guess that we can close this bug.

Thanks,

Rogério.

**Comment by dodev** [comment]
Hi!
The most recent changes committed to this repository did not fix the issues with GTK+ 3.12. I'll give it another try later today when I get back home.

About the maintainability - I have some ideas how this can be achieved. I'll ping you again on this topic.

**Comment by robertpainsi** [comment]
That would be awesome. Jump in here https://github.com/rbrito/pkg-mediterranean-gtk-themes/issues/19 for information about the refactoring and add your thoughts and improvement ideas. :smile_cat:

The gtk3-widget-factory, which shows all kind of ui elements, seems to have the same problem you posted. It can be installed by

```
sudo apt-get install gtk-3-examples #install
gtk3-widget-factory #run
```

I added this tool here because it's related to the issue but we should create a wiki page for these type of informaton. ping @rbrito :smirk_cat:

BTW: Your problem can be resolved by https://github.com/rbrito/pkg-mediterranean-gtk-themes/pull/28 Unfortunately this PR is just a quick and dirty c&p job and it's not tested well nor inspected. It also may change your titlebar edges to round corners.

**Comment by dodev** [comment]
Hey, Robert!

I've checked out #28 and it solves the transparent background of the title-bar. But the other widget problems are still present.

**Comment by robertpainsi** [comment]
Ohh, stupid me! Should have put on my glasses :joy_cat:
Now I see what the problem is in picture#2. And in picture#3 there are missing checkboxes right?
Let's see if the current refactoring process will fix these problems too.

**Comment by dodev** [comment]
Yep, the borders of the checkboxes are missing. Other things, that are not rendered correctly with the theme include: 
- missing radio button borders
- missing context menu borders
- the new GTK button groups are rendered as separate buttons.
  Here's a screenshot of the widget factory:
  ![screenshot from 2015-02-05 23 46 06](https://cloud.githubusercontent.com/assets/1612196/6068969/96461c00-ad91-11e4-895b-4b8bb08da49e.png)

As far as I understand, in GTK 3.12+ a lot of the widget's class names were changed, which rendered a lot of the CSS rules from Mediterranean\* without any matching elements. I was able to fix some of those in my fork (by sneaking in themes that work correctly with the latest versions of GTK), but I'm a man with a notorious lack of self-organization and I haven't achieved much.

The way to go, as I see it, is to create a template (#19) and start migrating the theme 1 rule at a time.

**Comment by robertpainsi** [comment]
Today I removed all my themes in my /usr/share/themes folder :scream_cat:. First I just wanted to delete all my downloaded themes I don't use anymore... then everything got out of hand and I deleted all of them. :pouting_cat: However...

I changed my theme to the White one and opened the widget-factory. I don't have the same problem with the radio buttons. The widgets look different too. In the MediterraneanWhite/gtk-3.0/assets folder you see the used icons for checkboxes and so on. Also the radio button border is in the asset image. So no css border would solve that (the easy way :smile_cat:)...

Btw: The MediterraneanWhite theme prints an error

```
(gtk3-widget-factory:16014): Gtk-WARNING **: Theme parsing error: unity.css:51:10: 'start' is not a valid property name
```

https://github.com/rbrito/pkg-mediterranean-gtk-themes/blob/master/MediterraneanWhite/gtk-3.0/unity.css#L49
has to be commented out. I won't to fix that for the master yet because I nearly finished my first common merge base stuff thingy :smirk_cat:
